AP_Periph: allow uploader.py to reboot AP_Periph

magicrub opened the pull request:
This allows uploader.py to trigger a reboot in AP_Periph when using ```./waf AP_Periph --upload```

This works as-is but it will work well with PR https://github.com/ArduPilot/ardupilot/pull/16033

khancyr commented on Tools/AP_Periph/AP_Periph.cpp:
```diff
@@ -265,9 +265,18 @@ void AP_Periph_FW::update()
         hal.rcout->set_serial_led_num_LEDs(HAL_PERIPH_NEOPIXEL_CHAN, HAL_PERIPH_NEOPIXEL_COUNT, AP_HAL::RCOutput::MODE_NEOPIXEL);
 #endif
+
+#ifdef HAL_PERIPH_ENABLE_RC_OUT
```
shouldn't this part of another pr ?

magicrub replied:
Whoops! Copy paste bug. I'll fix that.

magicrub commented:
I just rebased this on https://github.com/ArduPilot/ardupilot/pull/16033 so lets merge this after that one goes in.

magicrub commented:
note, this PR is just a single commit

tridge commented:
it would be better to recognise the "reboot" string

magicrub commented:
uploader.py is actually sending this:

    // These are the string definitions in uploader.py
    //            NSH_INIT        = bytearray(b'\x0d\x0d\x0d')
    //            NSH_REBOOT_BL   = b"reboot -b\n"
    //            NSH_REBOOT      = b"reboot\n"

    // This is the command sequence that is sent from uploader.py
    //            self.__send(uploader.NSH_INIT)
    //            self.__send(uploader.NSH_REBOOT_BL)
    //            self.__send(uploader.NSH_INIT)
    //            self.__send(uploader.NSH_REBOOT)

So I changed it to parse for:
```const char reboot_string[] = "\r\r\rreboot -b\n\r\r\rreboot\n";```

Works great and is a MUCH more specific string to be checking and therefor should never happen on accident. I also added HAL_PERIPH_LISTEN_FOR_SERIAL_UART_REBOOT_CMD_PORT to specify a serial port number and added the USB port to CubeOrange-periph. This is how it's documented in the hwdef.dat

```
# listen for reboot command from uploader.py script
# undefine to disable. Use -1 to allow on all ports, otherwise serial number index defined in SERIAL_ORDER starting at 0
define HAL_PERIPH_LISTEN_FOR_SERIAL_UART_REBOOT_CMD_PORT 0
```
